Add Elastic Agent 8.19.18 release notes (#15207)
elastic/elastic-agent: changelog/8.19.18.asciidoc

// begin 8.19.18 relnotes

[[release-notes-8.19.18]]
==  8.19.18

Review important information about the  8.19.18 release.

[discrete]
[[enhancements-8.19.18]]
=== Enhancements

> * Adds system.cpu.cores field to self monitoring data. https://github.com/elastic/elastic-agent/pull/14885[#14885] https://github.com/elastic/elastic-agent/issues/14862[#14862]

Reviewer: ```suggestion
* Adds `system.cpu.cores` field to self-monitoring data. https://github.com/elastic/elastic-agent/pull/14885[#14885] https://github.com/elastic/elastic-agent/issues/14862[#14862]
```

> * Don&#39;t forcibly inject the elasticdiagnostic extension when running in plain otel mode. https://github.com/elastic/elastic-agent/pull/15105[#15105] https://github.com/elastic/elastic-agent/issues/14606[#14606]

Reviewer: ```suggestion
* Don't forcibly inject the `elasticdiagnostic` extension when running in plain OTel mode. https://github.com/elastic/elastic-agent/pull/15105[#15105] https://github.com/elastic/elastic-agent/issues/14606[#14606]
```

[discrete]
[[bug-fixes-8.19.18]]
=== Bug fixes

> * Fix missing beat receiver trace logs from diagnostic bundle. https://github.com/elastic/elastic-agent/pull/14716[#14716] 

Reviewer: ```suggestion
* Fix an issue where Beat receiver trace logs were missing from the diagnostics bundle. https://github.com/elastic/elastic-agent/pull/14716[#14716]
```
